src/Test/Utilities/Portable/CommonTestBase.CompilationVerifier.cs

```diff
@@ -105,9 +105,14 @@ public void Emit(string expectedOutput, IEnumerable<ResourceDescription> manifes
                         MetadataSignatureUnitTestHelper.VerifyMemberSignatures(testEnvironment, expectedSignatures);
                     }
 
-                    if (expectedOutput != null)
+                    if (expectedOutput != null || expectedReturnCode != null)
                     {
-                        testEnvironment.Execute(mainModuleName, expectedOutput);
+                        var returnCode = testEnvironment.Execute(mainModuleName, args, expectedOutput);
```

Comment:
It does on the Desktop environment, but I needed to update @agocke's console capture code after the rebase.